src/Microsoft.OData.Edm/Vocabularies/Annotations/EdmTermExtensions.cs

```diff
@@ -0,0 +1,446 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="EdmTermExtensions.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using Microsoft.OData.Edm.Csdl;
+using System.Collections.Generic;
+using System.Linq;
+using System.IO;
+using System.Text;
+using Microsoft.OData.Edm.Csdl.Parsing.Ast;
+
+namespace Microsoft.OData.Edm.Vocabularies
+{
+    internal static class EdmTermExtensions
+    {
+        /// <summary>
+        /// Parses a <paramref name="defaultValue"/> into an IEdmExpression value of the correct EDM type.
+        /// </summary>
+        /// <param name=“typeReference”>The type of value.</param>
+        /// <param name=“defaultValue”>Original default value.</param>
+        /// <returns>An IEdmExpression of type <paramref name="typeReference".</paramref></returns>
+        public static IEdmExpression BuildDefaultEdmExpression(IEdmTypeReference typeReference, string defaultValue)
+        {
+            EdmTypeKind termTypeKind = typeReference.TypeKind();
+
+            // Create expressions/constants for the corresponding types
+            switch (termTypeKind) // TODO: Create test cases for all of these
+            {
+                case EdmTypeKind.Primitive:
+                    return BuildEdmPrimitiveValueExp(typeReference.AsPrimitive(), defaultValue);
+                case EdmTypeKind.Enum:
+                    // return BuildEdmEnumExp(typeReference.AsEnum(), defaultValue)
+                case EdmTypeKind.Complex:
+                case EdmTypeKind.Entity:
+                    // return BuildEdmStructuredTypeExp(typeReference.AsStructured(), defaultValue);
+                case EdmTypeKind.Collection:
+                    // return BuildEdmCollectionExp(typeReference, defaultValue);
+                case EdmTypeKind.Path:
+                    // return BuildEdmPathExp(typeReference.AsPath(), defaultValue);
+                case EdmTypeKind.TypeDefinition:
+                    // return BuildEdmTypeDefinitionExp(typeReference.AsTypeDefinition(), defaultValue);
+                    throw Error.NotImplemented();
+                case EdmTypeKind.EntityReference:
+                case EdmTypeKind.Untyped:
+                default:
+                    throw Error.NotSupported();

```

Comment:
>throw Error.NotSupported(); [](http://example.com/codeflow?start=20&length=27)

do you want it to throw an exception or return null?

note that, by spec, reading an annotation value should not throw an exception, so perhaps we should just return null here (and in line 44) and make sure we handle the null case in the caller.

Reply:
I see, gotcha! To double check, it would be reasonable to throw an exception in the caller in the case that the expression is null, but not when reading the annotation value itself?

Reply:
yes; programmatically creating it's fine to throw an exception, and it's fine for model validation to fail, but we shouldn't error when reading an invalid annotation.

Reply:
Done ✅